packages/devtools_app/lib/src/memory/flutter/memory_screen.dart

```diff
@@ -85,6 +85,16 @@ class MemoryBodyState extends State<MemoryBody> {
     if (newController == controller) return;
     controller = newController;
 
+    // Update the chart when the memorySource changes.
+    addAutoDisposeListener(controller.selectedSnapshotNotifier, () {
```

Comment:
instead of doing this wire-up here, make sure the controller properly tracks flowing through changes to selectedSnapshotNotifier itself only making this widget update when the new snapshot gets created.